pkg/edit/navigation_test.go

```diff
@@ -63,23 +66,102 @@ func TestNavigation_WidthRatio(t *testing.T) {
 	evals(f.Evaler, `@edit:navigation:width-ratio = 1 1 1`)
 	f.TTYCtrl.Inject(term.K('N', ui.Ctrl))
 	f.TestTTY(t,
-		"~"+string(os.PathSeparator)+"d> ", term.DotHere, "\n",
+		filepath.Join("~", "d"), "> ", term.DotHere, "\n",
 		" NAVIGATING  \n", Styles,
 		"************ ",
-		" d                a               ", Styles,
+		" d                a               \n", Styles,
 		"################ ++++++++++++++++ ",
+		"                  e              ", Styles,
+		"                 ////////////////",
 	)
 }
 
+// Test corner case: Inserting a selection when the CLI cursor is not at the
+// start of the edit buffer, but the preceding char is a space, does not
+// insert another space.
+func TestNavigationCornerCaseRedundantSpace(t *testing.T) {
+	f, cleanup := setupNav()
+	defer cleanup()
+
+	f.TTYCtrl.Inject(term.K('x'))              // user presses 'x'
+	f.TTYCtrl.Inject(term.K(' '))              // user presses Space
+	f.TTYCtrl.Inject(term.K('N', ui.Ctrl))     // begin navigation mode
+	f.TTYCtrl.Inject(term.K(ui.Down))          // select "e"
+	f.TTYCtrl.Inject(term.K(ui.Enter, ui.Alt)) // insert the "e" file name
+	f.TTYCtrl.Inject(term.K('z'))              // user presses 'z'
```

Comment:
This isn't really necessary, since the Alt-Enter press about would result in a buffer change; these kind of actions are only necessary when you are testing the *absense* of change.